### PR TITLE
[MWPW-128656] Template X receive Search Bar from Search Marquee

### PR DESCRIPTION
**Pull request by qiyundai**
Please always provide the [JIRA issue(s)](https://jira.corp.adobe.com/secure/RapidBoard.jspa?rapidView=34618) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix [MWPW-128656](https://jira.corp.adobe.com/browse/MWPW-128656)

**Description:**
Some small layout tweaking to avoid masonry col squeezing
Added sticky search bar if both **tool bar** and **search bar** are toggled on for template X block and the search marquee block is present on the page.

Test URLs:
- Before: https://stage--express--adobecom.hlx.page/express/templates/?lighthouse=on
- After: https://mwpw-128656--express--adobecom.hlx.page/express/templates/?lighthouse=on

### CODE REVIEW

**Review by vhargrave — CHANGES_REQUESTED** [review]
There are merge conflicts here currently

**Comment by toka1300** [comment]
Looks like the import reference for the fetchAPI is not pointing to the correct folder (linting error)

**Comment by toka1300** [comment]
When I filter on 'Free' or 'Premium', the filter goes back to 'Premium & Free'.
image

**Comment by toka1300** [comment]
When filtered on 'animated', are all of the templates supposed to animate on hover?

**Comment by qiyundai** [comment]
> When filtered on 'animated', are all of the templates supposed to animate on hover?

not necessarily. animated isn't really an accurate filter any more. Nonetheless, all of the templates you see with a small play icon on the bottom left cover are the ones with hover play effect.

**Comment by qiyundai** [comment]
> When I filter on 'Free' or 'Premium', the filter goes back to 'Premium & Free'. Same goes for the 'Static/Animated' filter. image

This is a new issue that exists before this PR as well. I left a fixme comment in the code and temporarily commented out the line of code causing the issue. This will not be a GA blocker but is worth a fast follow!

**Comment by toka1300** [comment]
> > When filtered on 'animated', are all of the templates supposed to animate on hover?
> 
> not necessarily. animated isn't really an accurate filter any more. Nonetheless, all of the templates you see with a small play icon on the bottom left cover are the ones with hover play effect.

The change you made seems to have fixed it, most I was seeing before were not animating, now most are.

**Comment by qiyundai** [comment]
> One other thing I just noticed and maybe not critical but the hover on the category list goes all the way down the page, making for a bit of a weird effect. Maybe should be sticky to top as you scroll down or make wrapper not go down the whole page. This is one I implemented, so I can take a look if you think it's worth it.
> 
> image

This is how it is on production now and Sean hasn't mentioned anything so I think we should leave it be. Thanks for the proactiveness, though!

File: express/blocks/template-x/template-x.js [hunk 1]
```diff
@@ -1400,6 +1412,149 @@ async function decorateBreadcrumbs(block) {
...
+        const clearBtn = searchWrapper.querySelector('.icon-search-clear');
+        const trendsContainer = searchWrapper.querySelector('.trends-container');
+        const suggestionsContainer = searchWrapper.querySelector('.suggestions-container');
+        const suggestionsList = searchWrapper.querySelector('.suggestions-list');
+
+        searchBar.addEventListener('click', (event) => {
+          event.stopPropagation();
+          searchWrapper.classList.remove('collapsed');
+          setTimeout(() => {
+            searchDropdown.classList.remove('hidden');
```
Comment (vhargrave) [inline express/blocks/template-x/template-x.js:1438]: setTimeout is usually a sign of code smell. Why are we waiting precisely 500 milliseconds? What's the goal here?

**Review by vhargrave — CHANGES_REQUESTED** [review]
Can you provide test links to test this sticky search bar?

**Review by vhargrave — CHANGES_REQUESTED** [review]
Additionally - you should probably add a unit test to test the search bar is being sticky right?

**Comment by qiyundai** [comment]
> Additionally - you should probably add a unit test to test the search bar is being sticky right?

Yes. Thanks for the reminder!

File: express/blocks/template-x/autocomplete-api-v3.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const url = 'https://adobesearch-atc.adobe.io/uss/v3/autocomplete';
+const experienceId = 'default-templates-autocomplete-v1';
+const scopeEntities = [
+  'HzTemplate',
+  // 'HzTextLockup', 'Icon', 'Photo', 'DesignAsset', 'Background'
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/blocks/template-x/autocomplete-api-v3.js:17]: @JingleH Any insight?

File: express/blocks/template-x/template-x.js (hunk 1 above)
Comment (qiyundai) [reply express/blocks/template-x/template-x.js:1438]: the goal was to let the expand animation of the search bar finish before the dropdown can appear.

File: express/blocks/template-x/autocomplete-api-v3.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (JingleH) [reply express/blocks/template-x/autocomplete-api-v3.js:17]: Yes we can remove that comment and reduce more bytes!

**Review by JingleH — APPROVED** [review]

**Review by apganapa — APPROVED** [review]

File: express/blocks/template-x/template-x.js (hunk 1 above)
Comment (vhargrave) [reply express/blocks/template-x/template-x.js:1437]: ok, that is definitely valid :)

**Review by vhargrave — DISMISSED** [review]
Looks good! Merge conflicts need to be fixed, but then we're good to go.

**Review by JingleH — APPROVED** [review]

**Review by apganapa — APPROVED** [review]

**Review by JingleH — APPROVED** [review]

**Pull request merged**